api/resourceserver/list_versioned_resource_types.go

```diff
@@ -0,0 +1,29 @@
+package resourceserver
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/concourse/atc/db"
+)
+
+func (s *Server) ListVersionedResourceTypes(pipeline db.Pipeline) http.Handler {
+	logger := s.logger.Session("list-versioned-resource-types")
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		resourcsTypes, err := pipeline.ResourceTypes()
+		if err != nil {
+			logger.Error("failed-to-get-resources-types", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		versionedResourceTypes := resourcsTypes.Deserialize()
+
+		w.WriteHeader(http.StatusOK)
```

Comment:
Mind setting the `Content-Type` before writing the header?

Reply:
done. Thank you!

Reply:
Thanks for actually adding it!